warehouse/macaroons/caveats.py

```diff
@@ -81,11 +81,14 @@ def __init__(self, macaroon, context, principals, permission):
         self.permission = permission
         self.verifier = pymacaroons.Verifier()
 
-    def verify(self, key):
+    def verify(self, key: str) -> bool:
         self.verifier.satisfy_general(V1Caveat(self))
+        self.verify_signature(key=key)
+        return True
 
+    def verify_signature(self, key: str) -> None:
         try:
-            return self.verifier.verify(self.macaroon, key)
+            self.verifier.verify(self.macaroon, key)
```

Comment:
```suggestion
            return self.verifier.verify(self.macaroon, key)
```
I'll agree that the pymacaroons API doesn't make a ton of sense here, but in case a future version begins returning `False` here, I think we should continue returning the result.

Reply:
Isn’t it more pythonic to return None for success, raise exception for problem?

Reply:
Sure, but in that case we'd still need to change this to raise an exception if `verify` ever returns false. Either works for me.

Reply:
Right, and one would hope that the project maintainer would have suitable version bump and changelog entries in that case!

Reply:
Yep, I’d much rather raise if False. "True or raise" feels to me
Line a receipe for a misunderstanding (like what happened there)

Reply:
I will be adding code to raise if False, but in parallel, I've opened https://github.com/ecordell/pymacaroons/issues/55 to try and officialize "raise or pass".

(I admit I'm not a huge fan of writing code imagining potential dependency API breakages, and I'm not a fan that in order to test this code we **will** have to monkeypatch and make the stub do things that cannot happen in real life. For all intended purposes we introducing new dead code :/ )

(Except if you think it's not worth it, I'll happily celebrate a change of mind :) )